### PR TITLE
Remove Coveralls (closes #3720)

### LINKED ISSUE #3720: Drop Coveralls?
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

I don't think Coveralls is checking things correctly. When we look at the CI results for commit https://github.com/fastify/fastify/pull/3712/commits/5a77a6b9a93d0b1e1c830a8c9bce997b2bcc7646 we can see that Coveralls is suggesting coverage remains at 100% for all OSes, but `tap` is saying otherwise:

https://coveralls.io/jobs/94867280
fastify - Vivaldi 2022-02-20 18-57-15

https://github.com/fastify/fastify/runs/5267302531?check_suite_focus=true
fastify - Vivaldi 2022-02-20 18-58-03

With #3717 and #3719, we will have coverage reports attached to the GitHub Actions panel itself. We'd have to download the artifacts to inspect what coverage is missing, instead of browsing a hosted website (Coveralls), but at least it'd be consistent and accurate(?).

### PR DESCRIPTION
**Pull request by jsumners**
As titled. Resolves #3720.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -136,7 +136,6 @@
     "ajv-merge-patch": "^5.0.1",
     "branch-comparer": "^1.1.0",
     "cors": "^2.8.5",
-    "coveralls": "^3.1.1",
```
Comment (Eomm) [inline package.json:139]: We need to remove the badge in the readme.md file as well

Comment (jsumners) [reply package.json:139]: Fixed.

**Review by Fdawgs — APPROVED** [review]
Goodnight sweet, CI blocking, Prince

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**